global.go

```diff
@@ -0,0 +1,248 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/gonum/matrix/mat64"
+)
+
+// GlobalMethod is a global optimizer. Typically will require more function
+// evaluations and no sense of local convergence
+type GlobalMethod interface {
+	// Need to tell the optimizer how many can be done at once (for example random restart)
+	InitGlobal(tasks int) // can this be concurrent bool? Seems like optimizer should be told how many tasks
+	// Global method may assume that the same task id always has the same pointer with it.
+	IterateGlobal(task int, loc *Location) (Operation, error)
+	Needs() struct{ Gradient, Hessian bool }
+}
+
+// Global uses a global optimizer to search for the gloabl minimum of a
+// function. A maximization problem can be transformed into a
+// minimization problem by multiplying the function by -1.
+//
+// Global, unlike Local, searches for the global minimum of the function.
+// GlobalMethods typically do not make any assumptions about the number and location
+// of local minima, and so the only convergence metric used is the convergence
+// of function values found at major iterations. GradientThreshold is not checked
+// during convergence, and initial values are not used.
+func Global(p Problem, dim int, settings *Settings, method GlobalMethod) (*Result, error) {
```

Comment:
It is natural but still nice that it is not possible to pass any initial guess here.